From d0ece156770a20a6c425da5da8948db154ce974f Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Wed, 6 Mar 2019 11:01:55 +0800
Subject: [PATCH 495/706] drm/imx: hdp: drop redundant drm_edid_to_eld() calls

drm_add_edid_modes() now fills in the ELD automatically, so the calls to
drm_edid_to_eld() are redundant. Remove them.

All the other places are obvious, but nv50 has detached
drm_edid_to_eld() from the drm_add_edid_modes() call.

Link: https://patchwork.freedesktop.org/patch/msgid/0959ca02b983afc9e74dd9acd190ba6e25f21678.1509545641.git.jani.nikula@intel.com

Zumeng Chen <zumeng.chen@windriver.com>

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index b3fc6413ba60..b19cfa15c273 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -700,9 +700,7 @@ static int imx_hdp_connector_get_modes(struct drm_connector *connector)
 				dev_dbg(hdp->dev, "use default video modes\n");
 				num_modes =
 					imx_hdp_default_video_modes(connector);
-			} else
-				/* Store the ELD */
-				drm_edid_to_eld(connector, edid);
+			}
 			kfree(edid);
 		} else {
 			dev_dbg(hdp->dev, "failed to get edid, ");
-- 
2.17.1

